core/src/com/unciv/json/HashMapVector2.kt

```diff
@@ -10,12 +10,14 @@ import java.util.HashMap
 class HashMapVector2<T> : HashMap<Vector2, T>() {
     companion object {
         init {
+            @Suppress("UNCHECKED_CAST") // kotlin can't tell that HashMapVector2 is also a MutableMap within generics
```

Comment:
"Honestly I've no fucking clue there" - you got it :+1: :+1: :+1:. Goes to show why justifying every single suppress is a good thing (and I just found a violation of that  elsewhere - whodunnit? me)